src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs

```diff
@@ -25,6 +23,17 @@ public static bool ShouldComputeIndex(Workspace workspace)
                 case WorkspaceKind.RemoteWorkspace:
                     // Always compute indices in the remote workspace.
                     return true;
+
+                case WorkspaceKind.Host:
+                case WorkspaceKind.MSBuild:
+                    // Compute indices in the host workspace when OOP is disabled.
+                    var remoteHostClientService = workspace.Services.GetService<IRemoteHostClientService>();
```

Comment:
why the way, this is only called inside of solution crawler and not every workspace register itself to solution crawler. is msbuild now register itself to solution crawler?

Reply:
> is msbuild now register itself to solution crawler?

No, it's covered by the commit message. AnalyzerRunner uses MSBuildWorkspace for targeted testing, and @Therzok and I are working on a new mode in it to run incremental analyzers.

Reply:
I think you can just do 

```csharp
public static bool ShouldComputeIndex(Workspace workspace)
{
      var remoteHostClientService = workspace.Services.GetService<IRemoteHostClientService>();
      if (remoteHostClientService is null)
                    {
                        return true;
                    }

                     return !remoteHostClientService.IsEnabled();
}
```